File: vulnerable-code/xss/vulnerable.php
<?php
/*
 * Cross-Site Scripting Demonstration
 *
 * INTENTIONALLY VULNERABLE CODE
 *
 * This example reflects user-controlled input directly into
 * HTML without output encoding.
 *
 * Educational and authorized laboratory use only.
 */

$name = $_GET['name'] ?? 'Guest';
?>

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Vulnerable XSS Example</title>
</head>
<body>

    <h1>Welcome</h1>

    <!-- VULNERABLE:
         User-controlled input is rendered directly into HTML. -->
    <p>Welcome, <?php echo $name; ?></p>

</body>
</html>

File: vulnerable-code/hardcoded-credentials/vulnerable.py
"""
Hardcoded Credentials Demonstration

INTENTIONALLY VULNERABLE CODE

This example embeds authentication secrets directly in source code.

Educational and authorized laboratory use only.
"""

USERNAME = "admin"
PASSWORD = "example-password"


def authenticate(username: str, password: str) -> bool:
    """
    Compare supplied credentials against values embedded in source code.
    """
    return username == USERNAME and password == PASSWORD


if __name__ == "__main__":
    supplied_user = input("Username: ")
    supplied_password = input("Password: ")

    if authenticate(supplied_user, supplied_password):
        print("Authentication successful.")
    else:
        print("Authentication failed.")
